src/js/PhotoSphereViewer.core.js

```diff
@@ -446,16 +449,34 @@ PhotoSphereViewer.prototype._reverseAutorotate = function() {
  * Remove a panorama image from the internal cache.
  * @param {String} panorama - The panorama uri, if missing, all the cache will bhe cleared.
  * @private
+ * @return {Boolean} False if cache is disabled, else otherwise.
  */
 PhotoSphereViewer.prototype._clearTexture = function(panorama) {
   if (true === this.config.cacheTextures) {
-    if(panorama && 'undefined' != typeof this.config.loadedTextures[panorama]){
-      delete this.config.loadedTextures[panorama];
-    } else {
-      delete this.config.loadedTextures[panorama];
-      this.config.loadedTextures = {};
+    if (panorama && 'undefined' != typeof this.prop.loadedTextures[panorama]) {
+      delete this.prop.loadedTextures[panorama];
+    }
+    else {
+      delete this.prop.loadedTextures[panorama];
+      this.prop.loadedTextures = {};
     }
     return true;
   }
   return false;
 };
+
+/**
+ * Check if the cache size has reached the limit
+ * @private
+ * @return null
+ */
+PhotoSphereViewer.prototype._normalizeCacheSize = function(pano, progressCallback) {
```

Comment:
This implementation is flawed, you remove only one item, whatever the number of extra items